integrations/azure-devops/azure_devops/client/azure_devops_client.py

```diff
@@ -163,6 +163,9 @@ async def generate_work_items(self) -> AsyncGenerator[list[dict[str, Any]], None
         """
         Retrieves a paginated list of work items within the Azure DevOps organization based on a WIQL query.
         """
+        selector = typing.cast(
+            AzureDevopsWorkItemResourceConfig, event.resource_config
+        ).selector
```

Comment:
client, shouldn't have access to the event context, lets move it outside and pass it as a param